packages/api/models/rules.js

```diff
@@ -93,19 +93,22 @@ class Rule extends Manager {
 
     if (!exists) {
       const err = {
-        message: 'Workflow doesn\'t exist'
+        message: `Workflow doesn\'t exist: s3://${bucket}/${key} for ${item.name}`
       };
       throw err;
     }
 
     const template = `s3://${bucket}/${key}`;
-    return {
+    const payload = {
       template,
       provider: item.provider,
-      collection: item.collection,
       meta: get(item, 'meta', {}),
       payload: get(item, 'payload', {})
     };
+
+    if (item.collection) payload.collection = item.collection;
+    if (item.rule) payload.rule = item;
```

Comment:
Should this be `payload.rule = item.rule;`?

Reply:
The item is the rule.  It may contain a collection but this is not required.  Intent is to pass the rule through as part of the payload.